blocks/algorithms/__init__.py

```diff
@@ -169,6 +170,9 @@ def __init__(self, step_rule=None, gradients=None, **kwargs):
                 zip(self.params, tensor.grad(self.cost, self.params))))
         self.step_rule = step_rule if step_rule else SteepestDescent()
 
+        self.total_gradient_norm = named_copy(
+            sum(g.norm(2) for g in self.gradients), "total_gradient_norm")
```

Comment:
Should probably be `tensor.sqrt(sum(g ** 2 for g in self.gradients))`; the L2 norm isn't additive.